src/main/java/lotto/domain/Lottos.java

```diff
@@ -4,17 +4,34 @@
 import java.util.Collections;
 import java.util.EnumMap;
 import java.util.List;
-import lotto.controller.generator.LottoGenerator;
+import lotto.domain.generator.LottoManualNumberGenerator;
+import lotto.domain.generator.LottoNumberGenerator;
 
 public class Lottos {
 
-    private final List<Lotto> lottoBunch = new ArrayList<>();
-    private final LottoGenerator lottoGenerator;
+    private final List<Lotto> lottoBunch;
 
-    public Lottos(LottoGenerator lottoGenerator, int purchasedLottoCount) {
-        this.lottoGenerator = lottoGenerator;
-        for (int i = 0; i < purchasedLottoCount; i++) {
-            lottoBunch.add(new Lotto(this.lottoGenerator.generateNumbers()));
+    public Lottos(LottoNumberGenerator lottoNumberGenerator, Piece purchasedLottoCount) {
+        lottoBunch = new ArrayList<>();
+        for (int i = 0; i < purchasedLottoCount.getPieceNumber(); i++) {
+            lottoBunch.add(new Lotto(lottoNumberGenerator.generateNumbers()));
+        }
+    }
+    
+    public Lottos mergeLottos(Lottos targetLottos) {
+        List<List<Number>> mergedLottoNumbers = new ArrayList<>();
+        mergeNumbers(mergedLottoNumbers, this);
+        mergeNumbers(mergedLottoNumbers, targetLottos);
+        LottoManualNumberGenerator mergedNumberGenerator =
+            new LottoManualNumberGenerator(mergedLottoNumbers);
+        int rawMergedPiece = mergedLottoNumbers.size();
+        Money sumMoney = new Money(rawMergedPiece * Lotto.PRICE);
+        return new Lottos(mergedNumberGenerator, new Piece(sumMoney, rawMergedPiece));
+    }
+
+    private void mergeNumbers(List<List<Number>> mergedLottoNumbers, Lottos lottos) {
+        for (Lotto lotto : lottos.getLottoBunch()) {
+            mergedLottoNumbers.add(lotto.getNumbers());
         }
```

Comment:
```suggestion
    private Lottos(final List<Lotto> lottoBunch) {
        this.lottoBunch = lottoBunch;
    }

    public Lottos(LottoNumberGenerator lottoNumberGenerator, int purchasedLottoCount) {
        final List<Lotto> lottoBunch = new ArrayList<>(purchasedLottoCount);
        for (int i = 0; i < purchasedLottoCount; i++) {
            lottoBunch.add(new Lotto(lottoNumberGenerator.generateNumbers()));
        }
        
        this(lottoBunch);
    }
    
    public Lottos mergeLottos(Lottos targetLottos) {
        final List<Lotto> lottos = new ArrayList(lottoBunch.size() + targetLottos.lottoBunch.size());
        lottos.addAll(lottoBunch);
        lottos.addAll(targetLottos.lottoBunch);
        
        return new Lottos(lottos);
    }
```

조금 더 간단하게 만들 수 있겠네요 :)